FIX: Special character alignment issue

keegangeorge opened the pull request:
This PR resolves an issue when editing tables with special characters next to empty boxes.

Previously, editing a table like this 
Column 1 | Column 2 | Column 3
-- | -- | --
| | editing | test

to change `editing` to `$`, would result in the table alignment being broken like so:

Previously, editing a table like this 
Column 1 | Column 2 | Column 3
-- | -- | --
 | $ | test

This is due to the `arrayToTable()` utility method taken from https://github.com/nijikokun/array-to-table generating tables without pipes surrounding the outer borders of the tables.

This PR uses a fork of `array-to-table`, found [here](https://github.com/Ygilany/array-to-table) that ensures tables are built with pipes around the borders. Along with that change the fork removes the optional alignment parameter that can be added to the method (however, we weren't using that anyways).

Now the tables will be built correctly like so:
| Column 1 | Column 2 | Column 3|
|-- | -- | --|
| | $ | test|

pmusaraj reviewed (commented):
Looks good. Can we add a small test to ensure that inputs and outputs from this arrayToTable helper function are what we expect? It might seem obvious now, but will be very valuable for any future refactors, will give the developer more confidence to make changes if there are tests.

keegangeorge commented:
> Can we add a small test to ensure that inputs and outputs from this arrayToTable helper function are what we expect?

@pmusaraj I've added the unit test in this next commit. However, I'm running into an issue with importing the `arrayToTable()` method like so:
```js
import { arrayToTable } from "../../../javascripts/discourse/lib/utilities";
```

It is giving me the following error:
<img width="1498" alt="Screen Shot 2022-08-10 at 3 44 35 PM" src="https://user-images.githubusercontent.com/30090424/184035162-e1f20774-d3c2-4359-987f-774367173beb.png">

I don't seem to see any issues with the import?

For now, I've added the test by duplicating the method code in under a helpers file and I am importing it like so:
```js
import arrayToTable from "../../helpers/arrayToTable";
```

which allows it to work correctly.

pmusaraj commented:
Good question, I wasn't sure how to do this either, but it looks like the library needs to be pulled out of `javascripts/discourse` and then it can be references. Pushed a commit.

pmusaraj commented on test/unit/lib/utilities-test.js:
```diff
@@ -0,0 +1,32 @@
+    assert.strictEqual(
+      arrayToTable(tableData),
+      markdownTableFixture,
+      "it creates a markdown table from an array of objects (with headers as keys)"
```
While we're at it, the test should probably check that special characters go into the correct column, right?

pmusaraj reviewed (approved):
Cool, LGTM!
